src/modules/api/user/link/link.controller.ts

```diff
@@ -1,7 +1,12 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Post, Body, Param } from '@nestjs/common';
 import { LinkService } from './link.service';
 
 @Controller('users/:userId/links')
 export class LinkController {
-  constructor(private linkService: LinkService) {}
+  constructor(private linkService: LinkService) { };
+  @Post()
```

Comment:
Good. But PUT is better approach for this. What do you think?

Reply:
yeah ! we can use the same route for edit also right ?

Reply:
why do you think so anna ?

Reply:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
Check 9.6

Reply:
Here, the resource is defined. Which is, users/:userId/link/:linkid so, we can go with PUT rather than POST

Reply:
Yeah anna ! got it .

Reply:
> Here, the resource is defined. Which is, users/:userId/link/:linkid so, we can go with PUT rather than POST

But the REST routes specifies a POST route to create a new entity right? and we wont have a :linkId at the moment of hitting this route, becase the object is not yet created.

Reply:
We can update with the same route if we pass /:linkid  , if it is a new resource /:linkid can be null right ?


> > Here, the resource is defined. Which is, users/:userId/link/:linkid so, we can go with PUT rather than POST
> 
> But the REST routes specifies a POST request to create a new entity right? and we wont have a :linkId at the moment of hitting this route, because the object is not yet created.

Reply:
> We can update with the same route if we pass /:linkid , if it is a new resource /:linkid can be null right ?
> 
> > > Here, the resource is defined. Which is, users/:userId/link/:linkid so, we can go with PUT rather than POST
> > 
> > 
> > But the REST routes specifies a POST request to create a new entity right? and we wont have a :linkId at the moment of hitting this route, because the object is not yet created.

Ah, you are right. Let this be POST. Because it's not Idempotent. 
I was thinking something else. 
Also, it's more about the nature of the response. If we maintain a single copy, we use PUT. If the instance keeps replicating, we use POST